assets/src/vo/date-time/datetime.js

```diff
@@ -0,0 +1,1156 @@
+	 * invalid.  This can happen when the moment instance is created with
+	 * invalid parameters.
+	 *
+	 * Note: with moment.timezone (which is the internal library),
+	 * moment.isValid() could return true, false or a string for why it's
+	 * invalid.  This is why a strict equality check is done for whether it is
+	 * true or not.
+	 *
+	 * @return {boolean}  True means the instance is invalid.
+	 */
+	isValid() {
+		return this[ privateProperties.datetime ].isValid() === true;
+	}
+
+	/**
+	 * Returns the difference between two DateTime instances as a Duration.
+	 *
+	 * @param {DateTime} otherDateTime
+	 * @return {Duration} An instance of Duration representing the difference
+	 * between the two DateTime objects.
+	 */
+	diff( otherDateTime ) {
+		DateTime.assertIsDateTime( otherDateTime );
+		return new Duration(
+			moment.duration(
+				this[ privateProperties.datetime ]
+					.diff( otherDateTime[ privateProperties.datetime ] )
+			)
+		);
+	}
+
+	/**
+	 * Returns the difference between this DateTime and "now" as a Duration.
+	 * @return {Duration} An instance of Duration representing the difference
+	 * between this DateTime and "now"
+	 */
+	diffNow() {
+		return new Duration(
+			moment.duration(
+				this[ privateProperties.datetime ]
+					.diff( moment() )
+			)
+		);
+	}
+
+	/**
+	 * Set the value of this DateTime to the end (i.e. the last millisecond) of
+	 * a unit of time.
+	 * @param {string} unit
+	 * @return {DateTime} Returns a new DateTime instance.
+	 */
+	endOf( unit ) {
+		return new DateTime(
+			this[ privateProperties.datetime ].clone().endOf( unit )
+		);
+	}
+
+	/**
+	 * Compares this DateTime with provided DateTime and returns whether they
+	 * are equal to each other.
+	 *
+	 * The two DateTimes are considered equal if they represent the same
+	 * millisecond, have the same zone and location, and are both valid.
+	 *
+	 * @param {DateTime} otherDateTime
+	 * @return {boolean}  True means they are equal
+	 */
+	equals( otherDateTime ) {
+		DateTime.assertIsDateTime( otherDateTime );
+		return this[ privateProperties.datetime ]
+			.isSame( otherDateTime[ privateProperties.datetime ] );
+	}
+
+	/**
+	 * Whether this DateTime is in the same unit of time as another DateTime
+	 *
+	 * eg. DateTime.fromLocal().hasSame( otherDT, 'day' ) //~> true if both the
+	 * same calendar day.
+	 *
+	 * Note: this will match all units equal or larger.  For example, passing in
+	 * `month` will check `month` and `year`.  So it's not only checking if the
+	 * two dates share the same month, but that they are the same month in the
+	 * same year.  If you passed in day, it would return whether the provided
+	 * DateTime is in the same day, month and year as this DateTime.
+	 *
+	 * @param {DateTime} otherDateTime
+	 * @param {string} unit
+	 * @return {boolean}  True means they are both in the same time for the
+	 * given unit.
+	 */
+	hasSame( otherDateTime, unit ) {
```

Comment:
I'm following the same api as Luxon: https://moment.github.io/luxon/docs/class/src/datetime.js~DateTime.html#instance-method-hasSame

I'd prefer to leave as is.  My guess is its the way it is because it follows the idea of a property always coming after the object (i.e. date2.day). For that reason, I think its more readable to have the object instance first since that's the frame of mind I'm in when I'm reading code.